### DISCUSSION TITLE
How to use Xmake to build godot-cpp project?

### DISCUSSION THREAD

**Discussion by OMObuan** [xmake-io/xmake#5198] [Q&A]
I have tried to use xrepo to use godot-cpp4 but it didn't work effectively, which will be unable to compile the official gdot-cpp example as listed:
![图片](https://github.com/xmake-io/xmake/assets/150120115/2d769776-9c7a-4ea3-9cb8-4e9f3ca86df7)
and the xmake.lua as listed:
<pre><code>
add_rules("mode.debug", "mode.release")
add_rules("plugin.compile_commands.autoupdate", { outputdir = "." })
set_languages("c++20")

add_requires("godotcpp4 4.2")

target("myfirst2dgamebycpp")
set_kind("shared")
add_packages("godotcpp4 4.2")
set_rundir("demo/bin")
set_warnings("everything")
add_files("src/*.cpp")
add_includedirs("src/")
</code></pre>
 Additionally, I used godot-cpp on github to write xmake.lua by myself, but no matter whether I use `add_files("godot-cpp/bin/*.a")` or try to link Sconstruct file in godot-cpp directory, it didn't work successfully. The Godot could not use the ".a" file as listed:
![图片](https://github.com/xmake-io/xmake/assets/150120115/09390e88-e34b-4bb8-a134-fb6c6d18992f)
and the xmake.lua as listed:
<pre><code>
add_rules("mode.debug", "mode.release")
add_rules("plugin.compile_commands.autoupdate", { outputdir = "." })
set_languages("c++20")

package("godotcpp")
set_sourcedir(path.join(os.scriptdir(), "godot-cpp"))
add_deps("scons")
on_install("linux", "windows", "macosx", "mingw", "cygwin", "iphoneos", "android", "msys", function(package)
	local configs = { "generate_bindings=yes" }
	table.insert(configs, "target=template_debug")
	table.insert(configs, "bits=64")
	table.insert(configs, "platform=linux")
	import("package.tools.scons").build(package, configs)
end)
package_end()

add_requires("godotcpp")

target("myfirst2dgamebycpp")
set_kind("shared")
add_packages("godotcpp")
set_rundir("demo/bin")
set_warnings("everything")
add_files("godot-cpp/bin/*.a")
add_files("src/*.cpp")
add_includedirs("src/")
add_includedirs("godot-cpp/gen/include")
add_includedirs("godot-cpp/include")
add_includedirs("godot-cpp/gdextension")
add_includedirs("godot-cpp/")
</code></pre>
Definitely, it is possible that I may be failed to write xmake.lua correctly. If possible, could anyone taught me how to wirte the file?

**Comment by star-hengxing** [comment]
`add_packages("godotcpp4 4.2")` -> `add_packages("godotcpp4")`

**Reply by OMObuan** [reply]
Thank you for answer! But unfortunately, I tried your method. But it still didn't work printing like this:
![图片](https://github.com/xmake-io/xmake/assets/150120115/c1e81379-6035-44c8-86c4-0eac46d848ff)

**Comment by star-hengxing** [comment]
Wait https://github.com/xmake-io/xmake-repo/pull/4290

**Reply by OMObuan** [reply]
OK. Thanks！

**Comment by star-hengxing** [comment]
Try it again. `xrepo update-repo`
